api/src/test/java/uk/gov/dhsc/htbhf/claimant/model/UpdatableClaimantFieldTest.java

```diff
@@ -0,0 +1,124 @@
+package uk.gov.dhsc.htbhf.claimant.model;
+
+import org.junit.jupiter.api.Test;
+import uk.gov.dhsc.htbhf.claimant.entity.Address;
+import uk.gov.dhsc.htbhf.claimant.entity.Claimant;
+
+import java.time.LocalDate;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.AddressTestDataFactory.aValidAddress;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.AddressTestDataFactory.aValidAddressBuilder;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.ClaimantTestDataFactory.*;
+
+class UpdatableClaimantFieldTest {
+
+    @Test
+    void shouldReportFieldsHaveDifferentValues() {
+        Claimant originalClaimant = Claimant.builder().build();
+        Claimant newClaimant = aClaimantWithExpectedDeliveryDate(LocalDate.now());
+
+        assertThat(UpdatableClaimantField.FIRST_NAME.valueIsDifferent(originalClaimant, newClaimant)).isTrue();
+        assertThat(UpdatableClaimantField.LAST_NAME.valueIsDifferent(originalClaimant, newClaimant)).isTrue();
+        assertThat(UpdatableClaimantField.DATE_OF_BIRTH.valueIsDifferent(originalClaimant, newClaimant)).isTrue();
+        assertThat(UpdatableClaimantField.EXPECTED_DELIVERY_DATE.valueIsDifferent(originalClaimant, newClaimant)).isTrue();
+        assertThat(UpdatableClaimantField.ADDRESS.valueIsDifferent(originalClaimant, newClaimant)).isTrue();
+    }
+
+    @Test
+    void shouldReportFieldsDoNotHaveDifferentValues() {
+        LocalDate expectedDeliveryDate = LocalDate.now();
+        Claimant originalClaimant = aClaimantWithExpectedDeliveryDate(expectedDeliveryDate);
+        Claimant newClaimant = aClaimantWithExpectedDeliveryDate(expectedDeliveryDate);
+
+        assertThat(UpdatableClaimantField.FIRST_NAME.valueIsDifferent(originalClaimant, newClaimant)).isFalse();
```

Comment:
Might it be worthwhile in these tests (especially this one where all the fields in the two objects are the same) to initially put an assertion to make sure the two objects aren't equal? Not essential but I thought it would just make it clear that the two objects aren't the same in all the tests.

Reply:
Last time we did that David raised (valid) questions about having an assert in the given step

Reply:
Which I disagreed with last time too, I think its valid to assert that your input data is in the correct state before you start the test. Maybe you could also validly say that it should be in the then step as it is an assertion.

Reply:
I've added the assert